sdk/maps/azure-maps-timezone/azure/maps/timezone/aio/_timezone_client_async.py

```diff
@@ -0,0 +1,303 @@
+            :dedent: 4
+            :caption: Creating the MapsTimezoneClient with a token credential.
+    """
+
+    def __init__(
+        self,
+        credential: Union[AzureKeyCredential, AsyncTokenCredential],
+        **kwargs: Any
+    )-> None:
+
+        self._maps_client = _MapsTimezoneClient(
+            credential=credential,
+            api_version=kwargs.pop("api_version", VERSION),
+            authentication_policy=kwargs.pop(
+                "authentication_policy", _authentication_policy(credential)),
+            **kwargs
+        )
+        self._timezone_client = self._maps_client.timezone
+
+    async def close(self)-> None:
+        await self._maps_client.close()
+
+    async def __aenter__(self):
+        await self._maps_client.__aenter__()
+        return self
+
+    async def __aexit__(self, *args):
+        return await self._maps_client.__aexit__(*args)
+
+    @distributed_trace_async
+    async def get_timezone_by_id(
+        self,
+        timezone_id: str,
+        ** kwargs: Any
+    )-> TimezoneResult:
+        """Returns current, historical, and future time zone information for the specified IANA
+        time zone ID.
+
+        :param timezone_id: The IANA time zone ID. Required. For example: "Europe/Berlin".
+        :type timezone_id: str
+        :keyword accept_language: Specifies the language code in which the timezone names should be
+         returned. If no language code is provided, the response will be in "EN". Please refer to
+         `Supported Languages <https://docs.microsoft.com/en-us/azure/azure-maps/supported-languages>`
+         for details. Default value is None.
+        :type accept_language: str
+        :keyword options: Alternatively, use alias "o". Options available for types of information
+         returned in the result. Known values are: "none", "zoneInfo", "transitions", and "all". Default
+         value is None.
+        :type options: str or ~azure.maps.timezone.models.TimezoneOptions
+        :keyword time_stamp: Alternatively, use alias "stamp", or "s". Reference time, if omitted, the
+         API will use the machine time serving the request. Default value is None.
+        :type time_stamp: ~datetime.datetime
+        :keyword daylight_savings_time_from: Alternatively, use alias "tf". The start date from which
+         daylight savings time (DST) transitions are requested, only applies when "options" = all or
+         "options" = transitions. Default value is None.
+        :type daylight_savings_time_from: ~datetime.datetime
+        :keyword daylight_savings_time_lasting_years: Alternatively, use alias "ty". The number of years
+         from "transitionsFrom" for which DST transitions are requested, only applies when "options" =
+         all or "options" = transitions. Default value is None.
+        :type daylight_savings_time_lasting_years: int
+        :keyword callable cls: A custom type or function that will be passed the direct response
```

Comment:
Are we adding `cls` to our external doc string? Maybe we can confirm with architect to see if we need these for our customers.